ui/raidboss/data/06-ew/trial/endsinger.ts

```diff
@@ -0,0 +1,251 @@
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6652', source: 'The Endsinger', capture: true }),
+      response: Responses.tankCleave(),
+    },
+    {
+      id: 'Endsinger Epigonoi',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6646', source: 'The Endsinger', capture: true }),
+      condition: (_data, matches) => {
+        return matches.x !== '100' && matches.y !== '100';
+      },
+      suppressSeconds: 3,
+      infoText: (_data, matches, output) => {
+        if (matches.x === '100' || matches.y === '100')
+          return output.intercardinal?.();
+
+        return output.cardinal?.();
+      },
+      outputStrings: {
+        cardinal: {
+          en: 'Cardinal edge',
+        },
+        intercardinal: {
+          en: 'Intercardinal edge',
+        },
+      },
+    },
+    {
+      id: 'Endsinger Interstellar',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '664D', source: 'The Endsinger', capture: true }),
+      alertText: (_data, matches, output) => {
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+
+        // Handle cardinals the easy way
+        if (x === 100) {
+          return output.direction!({
+            dir1: output.east!(),
+            dir2: output.west!(),
+          });
+        }
+        if (y === 100) {
+          return output.direction!({
+            dir1: output.north!(),
+            dir2: output.south!(),
+          });
+        }
+
+        if (x < 100) {
+          if (y < 100) {
+            return output.direction!({
+              dir1: output.northeast!(),
+              dir2: output.southwest!(),
+            });
+          }
+          return output.direction!({
+            dir1: output.northwest!(),
+            dir2: output.southeast!(),
+          });
+        }
+        if (y < 100) {
+          return output.direction!({
+            dir1: output.northwest!(),
+            dir2: output.southeast!(),
+          });
+        }
+        return output.direction!({
+          dir1: output.northeast!(),
+          dir2: output.southwest!(),
+        });
+      },
+      outputStrings: {
+        north: Outputs.north,
+        northeast: Outputs.northeast,
+        east: Outputs.east,
+        southeast: Outputs.southeast,
+        south: Outputs.south,
+        southwest: Outputs.southwest,
+        west: Outputs.west,
+        northwest: Outputs.northwest,
+        direction: {
+          en: '${dir1} / ${dir2}',
+        },
+      },
+    },
+    {
+      id: 'Endsinger Nemesis',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '664E', source: 'The Endsinger', capture: true }),
+      condition: Conditions.targetIsYou(),
+      response: Responses.spread(),
+    },
+    {
+      id: 'Endsinger Ultimate Fate',
+      type: 'GameLog',
+      netRegex: NetRegexes.gameLog({ code: '0039', line: '4 seconds until oblivion is sealed\\.\\.\\.' }),
```

Comment:
What about making this Ultimate Fate (6B59) starts casting? In general I would prefer to avoid game logs as much as possible.  It's extra translation work, and there's no end of people who don't have game logs for various reasons.

```
00|2021-12-08T18:04:27.0000000-08:00|0039||4 seconds until oblivion is sealed...|bab2a4e54fc622fb
20|2021-12-08T18:04:27.9420000-08:00|4000F88A|The Endsinger|6B59|Ultimate Fate|4000F88A|The Endsinger|4.70|100.00|100.00|0.00|-3.14|92bb7e981862f743
```

(Also, CI is complaining about missing `capture: false` on this line)

Reply:
I tried this with startsCasting and was having issues with it. I'll revisit it tomorrow, probably was a PEBKAC error.